packages/flutter/lib/src/material/text_field.dart

```diff
@@ -835,6 +861,32 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
   FocusNode? _focusNode;
   FocusNode get _effectiveFocusNode => widget.focusNode ?? (_focusNode ??= FocusNode());
 
+  MaxLengthEnforcement? _maxLengthEnforcement;
+  MaxLengthEnforcement get _effectiveMaxLengthEnforcement {
```

Comment:
They are slightly different for now, since they use different judge condition: `defaultTargetPlatform` and `Theme.of(context).platform` now. It will be better if they use the same, and take it out to a static method. What do you think?

Reply:
Oh I missed the `(TargetPlatform platform)`